chainer/functions/array/split_axis.py

```diff
@@ -14,31 +15,67 @@ class SplitAxis(function_node.FunctionNode):
     """Function that splits multiple arrays along the specified axis."""
 
     def __init__(self, indices_or_sections, axis):
-        if not isinstance(
-                indices_or_sections,
-                six.integer_types + (collections.Iterable,)):
-            raise TypeError('indices_or_sections must be integer or 1-D array')
-        if (chainer.is_debug() and
-                isinstance(indices_or_sections, collections.Iterable)):
-            for p, n in six.moves.zip(
-                    indices_or_sections, indices_or_sections[1:]):
+        indices, sections = self._get_indices_or_sections(indices_or_sections)
+        assert (indices is None) != (sections is None)
+        self.indices = indices
+        self.sections = sections
+        self.axis = axis
+
+    def _get_indices_or_sections(self, indices_or_sections):
```

Comment:
`self` is not used.
How about using module level function?

Reply:
Fixed.